pkg/solana/solana_service.go

```diff
@@ -0,0 +1,536 @@
+	if err != nil {
+		return nil, fmt.Errorf("failed to get account info: %w", err)
+	}
+
+	return convertAccountResult(account), nil
+}
+
+func (ss *solanaService) GetBalance(ctx context.Context, req commonsol.GetBalanceRequest) (*commonsol.GetBalanceReply, error) {
+	reader, err := ss.chain.Reader()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get reader: %w", err)
+	}
+
+	balance, err := reader.BalanceWithCommitment(ctx, solana.PublicKey(req.Addr), rpc.CommitmentType(req.Commitment))
+	if err != nil {
+		return nil, fmt.Errorf("failed to get balance: %w", err)
+	}
+
+	return &commonsol.GetBalanceReply{
+		Value: balance,
+	}, nil
+}
+
+func (ss *solanaService) SimulateTX(ctx context.Context, req commonsol.SimulateTXRequest) (*commonsol.SimulateTXReply, error) {
+	tx, err := solana.TransactionFromBase64(req.EncodedTransaction)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode transaction: %w", err)
+	}
+	accounts := &rpc.SimulateTransactionAccountsOpts{
+		Encoding:  solana.EncodingType(req.Opts.Accounts.Encoding),
+		Addresses: make([]solana.PublicKey, len(req.Opts.Accounts.Addresses)),
+	}
+	for _, addr := range req.Opts.Accounts.Addresses {
+		accounts.Addresses = append(accounts.Addresses, solana.PublicKey(addr))
+	}
+
+	res, err := ss.chain.MultiClient().SimulateTx(ctx, tx, &rpc.SimulateTransactionOpts{
+		SigVerify:              req.Opts.SigVerify,
+		Commitment:             rpc.CommitmentType(req.Opts.Commitment),
+		ReplaceRecentBlockhash: req.Opts.ReplaceRecentBlockhash,
+		Accounts:               accounts,
+	})
+	if err != nil {
+		return nil, fmt.Errorf("simulate tx failed: %w", err)
+	}
+	var simErr string
+	if res.Err != nil {
+		simErr = fmt.Sprintf("%v", res.Err)
+	}
+	return &commonsol.SimulateTXReply{
+		Err:           simErr,
+		Logs:          res.Logs,
+		Accounts:      convertAccounts(res.Accounts),
+		UnitsConsumed: res.UnitsConsumed,
+	}, nil
+}
+
+func (ss *solanaService) RegisterLogTracking(ctx context.Context, req commonsol.LPFilterQuery) error {
+	lp := ss.chain.LogPoller()
+	if lp.HasFilter(ctx, req.Name) {
+		return nil
+	}
+
+	f, err := convertFilter(req)
+	if err != nil {
+		return err
+	}
+
+	err = lp.RegisterFilter(ctx, f)
+	if err != nil {
+		return fmt.Errorf("failed to register fitler: %w", err)
+	}
+
+	return nil
+}
+
+func (ss *solanaService) UnregisterLogTracking(ctx context.Context, filterName string) error {
+	lp := ss.chain.LogPoller()
+	if !lp.HasFilter(ctx, filterName) {
+		return nil
+	}
+
+	return lp.UnregisterFilter(ctx, filterName)
+}
+
+func (ss *solanaService) QueryTrackedLogs(ctx context.Context, filterQuery []query.Expression,
+	limitAndSort query.LimitAndSort) ([]*commonsol.Log, error) {
+	lp := ss.chain.LogPoller()
+	// TODO derive query name from filterQuery
+	logs, err := lp.FilteredLogs(ctx, filterQuery, limitAndSort, "")
+	if err != nil {
+		return nil, fmt.Errorf("failed to filter logs: %w", err)
+	}
+
+	res := make([]*commonsol.Log, len(logs))
```

Comment:
Found a bug with this while working on my integration tests: 
```suggestion
	res := make([]*commonsol.Log, 0, len(logs))
```